File: limbo/library/src/main/java/co/windly/limbo/fragment/base/LimboFragmentPresenter.java
package co.windly.limbo.fragment.base;

import android.content.Context;
import androidx.annotation.NonNull;
import com.hannesdorfmann.mosby3.mvp.MvpBasePresenter;
import io.reactivex.disposables.CompositeDisposable;
import io.reactivex.disposables.Disposable;

public abstract class LimboFragmentPresenter<V extends LimboFragmentView> extends MvpBasePresenter<V> {

  //region Threading

  protected CompositeDisposable disposables = new CompositeDisposable();

  public boolean addDisposable(@NonNull Disposable disposable) {
    return disposables.add(disposable);
  }

  public void clearDisposables() {
    disposables.clear();
  }

  //endregion

  //region Managers

  public void initializeManagers(@NonNull Context context) {
    // No-op.
  }

  //endregion
}

File: limbo/library/src/main/java/co/windly/limbo/fragment/base/LimboFragmentPresenter.kt
package co.windly.limbo.fragment.base

import com.hannesdorfmann.mosby3.mvp.MvpBasePresenter
import io.reactivex.disposables.CompositeDisposable
import io.reactivex.disposables.Disposable

abstract class LimboFragmentPresenter<V : LimboFragmentView> : MvpBasePresenter<V>() {

  //region Reactive

  protected val disposables: CompositeDisposable
    by lazy { CompositeDisposable() }

  fun addDisposable(disposable: Disposable): Boolean =
    disposables.add(disposable)

  fun clearDisposables() {
    disposables.clear()
  }

  //endregion
}
